test/end-to-end/auth-client.js

```diff
@@ -0,0 +1,76 @@
+'use strict';
+
+const _ = require('lodash');
+const nightwatch = require('nightwatch');
+const test = require('ava');
+
+const smartcar = require('../../');
+
+const {getAuthClientParams, runAuthFlow} = require('./helpers');
+const config = require('../config');
+
+const context = {};
+
+test.before(() => {
+  context.client = nightwatch.initClient(config.get('nightwatch'));
+  context.browser = context.client.api();
+});
+
+test('exchangeCode', (t) => {
+  const client = new smartcar.AuthClient(getAuthClientParams());
+
+  const authUrl = client.getAuthUrl();
+
+  return runAuthFlow(context.client, context.browser, authUrl)
```

Comment:
nit: you could just make the test an async function, mixing the two is a bit weird imo...